tests/test_cloud_api.py

```diff
@@ -390,6 +453,173 @@ def test_get_servers_success_filtered(self, mock_popen):
+            "rb",
+        ) as f:
+            file = f.read()
+            mock_process_sites = self.create_mock_process(0, file)
+
+        mock_popen.return_value = mock_process_sites
+
+        with self.assertRaises(ValueError) as e:
+            self.cloud_api.get_meo_sites(
+                "13d225e3-70cd-4430-9866-e08c8ae02662", "badsite"
+            )
+            self.assertEqual(str(e.exception), "Site(s) not found: badsite")
+
+    #########################################
+    # get_meo_site_instances method
+    #########################################
+
+    @patch("subprocess.Popen")
+    def test_get_site_instances_all_success(self, mock_popen):
+        """
+        Test for success trying to get a list of site instances for a MEO environment
+        """
+
+        with open(
+            os.path.join(
+                self.cwd, "mocks/cloud_api/api_codebases_environments-list.json"
+            ),
+            "rb",
+        ) as f:
+            file = f.read()
+            mock_process_environments = self.create_mock_process(0, file)
+        with open(
+            os.path.join(self.cwd, "mocks/cloud_api/api_environments_sites-list.json"),
+            "rb",
+        ) as f:
+            file = f.read()
+            mock_process_sites = self.create_mock_process(0, file)
+
+        mock_popen.side_effect = [
+            mock_process_environments,
+            mock_process_sites,
+        ]
+
+        site_instances, environment = self.cloud_api.get_meo_site_instances(
+            "13d225e3-70cd-4430-9866-e08c8ae02662", "Prod"
+        )
+        self.assertEqual(len(site_instances), 3)
+        self.assertEqual(environment, "b700df4a-ca0d-4372-9f5d-2960c5745930")
+
+    @patch("subprocess.Popen")
+    def test_get_site_instances_one_success(self, mock_popen):
+        """
+        Test for success trying to get a single site instance for a MEO environment
+        """
+
+        with open(
+            os.path.join(
+                self.cwd, "mocks/cloud_api/api_codebases_environments-list.json"
+            ),
+            "rb",
+        ) as f:
+            file = f.read()
+            mock_process_environments = self.create_mock_process(0, file)
+        with open(
+            os.path.join(self.cwd, "mocks/cloud_api/api_environments_sites-list.json"),
+            "rb",
+        ) as f:
+            file = f.read()
+            mock_process_sites = self.create_mock_process(0, file)
+
+        mock_popen.side_effect = [
+            mock_process_environments,
+            mock_process_sites,
+        ]
+
+        site_instances, _ = self.cloud_api.get_meo_site_instances(
+            "13d225e3-70cd-4430-9866-e08c8ae02662", "Prod", "site2"
+        )
+        self.assertEqual(len(site_instances), 1)
+        self.assertEqual(
+            site_instances.get("site2"), "1b70caa3-c4f9-40df-848f-e76d4c6a31d7"
+        )
+
+    @patch("subprocess.Popen")
+    def test_get_site_instances_two_failure(self, mock_popen):
```

Comment:
Is this the same as `test_get_sites_two_failure`?  The names and ordering make it difficult to tell.